sqlglot/optimizer/qualify_columns.py

```diff
@@ -40,7 +40,7 @@ def qualify_columns(
     """
     schema = ensure_schema(schema)
     infer_schema = schema.empty if infer_schema is None else infer_schema
-    pseudocolumns = Dialect.get_or_raise(schema.dialect).PSEUDOCOLUMNS
+    pseudocolumns = Dialect.get_or_raise(schema.dialect, instance=False).PSEUDOCOLUMNS
```

Comment:
schema.dialect should already be an instance so we don't have to reinstantiate it